pigpen-cascading/src/main/java/pigpen/cascading/SingleIterationSeq.java

```diff
@@ -53,16 +63,24 @@ public Object first() {
   }
 
   public ISeq next() {
-    if (state.isRealized == false) {
+    if (state.rest == state) {
```

Comment:
I was wondering the same. The code is an adaptation of clojure's IteratorSeq, which is what I was using originally but caused OOMEs. If you think it's safe to assume PigPen users will never want to use multi-threaded access from their UDFs, I'm happy to remove the synchronization.